queasars/minimum_eigensolvers/base/termination_criteria.py

```diff
@@ -42,34 +42,48 @@ class BestIndividualChangeTolerance(EvolvingAnsatzMinimumEigensolverBaseTerminat
     :param minimum_change: absolute expectation value improvement below which the algorithm shall terminate.
         Must be at least 0
     :type minimum_change: float
+    :param allowed_consecutive_violations: determines how often the threshold value can be violated consecutively
+        before this criterion chooses to terminate. If set to 0, this terminates the first time the change falls
+        below the threshold value. If set to 2 for example, this terminates the first time the change falls below
+        the threshold three consecutive times. Must be at least 0.
+    :type allowed_consecutive_violations: int
     """
 
-    def __init__(self, minimum_change: float):
+    def __init__(self, minimum_change: float, allowed_consecutive_violations: int = 0):
         if minimum_change <= 0:
             raise ValueError("The minimum absolute improvement parameter must be bigger than 0!")
+        if allowed_consecutive_violations < 0:
+            raise ValueError("allowed_consecutive_violations must be at least 0!")
 
         self._minimum_change: float = minimum_change
-        self._previous_expectation_value: float = float("inf")
+        self._allowed_consecutive_violations: int = allowed_consecutive_violations
+        self._previous_expectation_value: Optional[float] = None
+        self._change_history: list[float] = []
 
     def reset_state(self) -> None:
-        self._previous_expectation_value = float("inf")
+        self._change_history = []
 
     def check_termination(
         self,
         population_evaluation: BasePopulationEvaluationResult,
         best_individual: BaseIndividual,
         best_expectation_value: float,
     ) -> bool:
-        if self._previous_expectation_value == float("inf"):
+
+        if self._previous_expectation_value is None:
             self._previous_expectation_value = population_evaluation.best_expectation_value
             return False
 
-        change: float = abs(self._previous_expectation_value - population_evaluation.best_expectation_value)
-        if change >= self._minimum_change:
-            self._previous_expectation_value = population_evaluation.best_expectation_value
+        change = abs(self._previous_expectation_value - population_evaluation.best_expectation_value)
+        self._change_history.append(change)
+        self._previous_expectation_value = population_evaluation.best_expectation_value
+
+        if len(self._change_history) < self._allowed_consecutive_violations + 1:
             return False
 
-        return True
+        max_change = max(self._change_history[-self._allowed_consecutive_violations - 1 :])
+
+        return not max_change >= self._minimum_change
```

Comment:
Maybe rename `max_change` to `max_change_in_last_relevant_window` or something similar? And use `return max_change_in_last_relevant_window < self._minimum_change`?

Reply:
You are right, that is much more readable. I adjusted it accordingly.